Core/Core/Api/GraphQL/Client.cs

```diff
@@ -23,66 +26,46 @@
 
 namespace Speckle.Core.Api;
 
-public sealed partial class Client : IDisposable
+public sealed partial class Client : ISpeckleGraphQLClient, ISpeckleGraphQLSubscriber, IDisposable
 {
-  [Obsolete]
-  internal Client() { }
-
-  public Client(Account account)
-  {
-    Account = account ?? throw new SpeckleException("Provided account is null.");
-
-    HttpClient = Http.GetHttpProxyClient(null, TimeSpan.FromSeconds(30));
-    Http.AddAuthHeader(HttpClient, account.token);
-
-    HttpClient.DefaultRequestHeaders.Add("apollographql-client-name", Setup.HostApplication);
-    HttpClient.DefaultRequestHeaders.Add(
-      "apollographql-client-version",
-      Assembly.GetExecutingAssembly().GetName().Version.ToString()
-    );
-
-    GQLClient = new GraphQLHttpClient(
-      new GraphQLHttpClientOptions
-      {
-        EndPoint = new Uri(new Uri(account.serverInfo.url), "/graphql"),
-        UseWebSocketForQueriesAndMutations = false,
-        WebSocketProtocol = "graphql-ws",
-        ConfigureWebSocketConnectionInitPayload = _ =>
-        {
-          return Http.CanAddAuth(account.token, out string? authValue) ? new { Authorization = authValue } : null;
-        },
-      },
-      new NewtonsoftJsonSerializer(),
-      HttpClient
-    );
-
-    GQLClient.WebSocketReceiveErrors.Subscribe(e =>
-    {
-      if (e is WebSocketException we)
-      {
-        Console.WriteLine(
-          $"WebSocketException: {we.Message} (WebSocketError {we.WebSocketErrorCode}, ErrorCode {we.ErrorCode}, NativeErrorCode {we.NativeErrorCode}"
-        );
-      }
-      else
-      {
-        Console.WriteLine($"Exception in websocket receive stream: {e}");
-      }
-    });
-  }
+  public ProjectResource Project { get; }
+  public ModelResource Model { get; }
+  public VersionResource Version { get; }
+  public ActiveUserResource ActiveUser { get; }
+  public OtherUserResource OtherUser { get; }
+  public ProjectInviteResource ProjectInvite { get; }
+  public CommentResource Comment { get; }
 
   public string ServerUrl => Account.serverInfo.url;
 
   public string ApiToken => Account.token;
 
-  public System.Version? ServerVersion { get; set; }
+  public System.Version? ServerVersion { get; private set; }
 
   [JsonIgnore]
-  public Account Account { get; set; }
+  public Account Account { get; }
+  private HttpClient HttpClient { get; }
 
-  private HttpClient HttpClient { get; set; }
+  public GraphQLHttpClient GQLClient { get; }
 
-  public GraphQLHttpClient GQLClient { get; set; }
+  /// <param name="account"></param>
+  /// <exception cref="ArgumentException"><paramref name="account"/> was null</exception>
+  public Client(Account account)
+  {
+    Account = account ?? throw new ArgumentException("Provided account is null.");
+
+    Project = new(this);
```

Comment:
Its all an internal implementation detail for now... Tho, a pattern like this could be useful for having some parts of core that are designed to be consumed by consumers in a DI container, and others who are not and just want a class like the `Client` to setup everything for them...